services/visual_recognition/v3.js

```diff
@@ -62,301 +37,331 @@ module.exports = function (RED) {
+            params['threshold']=msg.params['threshold'];
+          }
+          cb();
+        });
+      }); 
+    } else if (urlCheck(msg.payload)) {
+      params['url'] = msg.payload;
+      if (msg.params != null && msg.params.classifier_ids != null) {
+        params['classifier_ids']=msg.params['classifier_ids'];
       }
+      if (msg.params != null && msg.params.owners != null) {
+        params['owners']=msg.params['owners'];
+      }
+      if (msg.params != null && msg.params.threshold != null) {
+        params['threshold']=msg.params['threshold'];
+      }
+      return cb();
+    } else {
+      node.status({fill:'red', shape:'ring', text:'payload is invalid'});          
+      node.error('Payload must be either an image buffer or a string representing a url', msg);
+    }
+  }
 
-      var params = {};
-
-      // This is the callback after the call to the watson service.    
-      // Set up as a var within this scope, so it has access to node, msg etc. 
-      // in preparation for the Watson service action       
-      var actionComplete = function(err, body, other) {
 
-        if (err != null && body==null)
-        {
+  function addTask (asyncTasks, msg, key, listParams, node) {
+    asyncTasks.push(function (callback) {
+      var buffer = msg.params[key];
+      temp.open({suffix: '.' + fileType(buffer).ext}, function (err, info) {
+        if (err) {
           node.status({fill:'red', shape:'ring', 
-                       text:'call to watson visual recognition v3 service failed'}); 
-          msg.result = {};
-          msg.result['error_code']= err.code;
-          if (!err.error)
-            msg.result['error']= err.error;
-          console.log('Error:', err.code);
-          node.error('Error code : ' +  err.code);
-          return;
-        }
-        else if (body.images[0].error)
-        {
-          var err_desc = body.images[0].error.description;
-          var err_id = body.images[0].error.error_id;
-          node.status({fill:'red', shape:'ring', 
-                       text:'call to watson visual recognition v3 service failed'}); 
-          msg.result = {};
-          msg.result['error_id']= err_id;
-          msg.result['error']= err_desc;
-          console.log('Error:', err_desc);
-          msg.payload='see msg.result.error';
-          node.send(msg); 
-        }
-        else {
-          msg.result = {};
-          msg.result['all'] = body;
-          msg.payload='see msg.result'; // to remove any Buffer that could remains
-          node.send(msg); 
-          node.status({});
-        }
-      }; // actionComplete    
-
-      var actionComplete2 = function(err, body, other) {
+                       text:'unable to open image stream'});          
+          node.error('Node has been unable to open the image stream', msg);
+          return callback('open error on '+key);
+        }  
+        stream_buffer(info.path, msg.params[key], function () {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'stream_buffer' is not in camel case.](https://www.codacy.com/app/BetaWorks-NodeRED-Watson/node-red-node-watson/file/2737047177/issues/source?bid=3330694&fileBranchId=3352492#l188)